lib/Runtime/Math/JavascriptMath.cpp

```diff
@@ -901,9 +901,9 @@ namespace Js
             ScriptContext * scriptContext = function->GetScriptContext();
 
             TypeId typeId = JavascriptOperators::GetTypeId(arrayArg);
-            if (!JavascriptNativeArray::Is(typeId) && !(TypedArrayBase::Is(typeId) && typeId != TypeIds_CharArray && typeId != TypeIds_BoolArray))
```

Comment:
is this change in semantics? or were the extra conditions just redundant?

Reply:
TypedArrayBase::Is does a range check for typeId and TypeIds_CharArray and TypeIds_BoolArray fall in that range. Presumably, they didn't fall in the range when this code was written.

Reply:
If `TypedArrayBase::Is(typeId)` returns true for `TypeIds_CharArray`, then this is a change in semantics: the logic before (pushing in the negation) was
```cpp
!JavascriptNativeArray::Is(typeId) &&
  (!TypedArrayBase::Is(typeId) || typeId == TypeIds_CharArray || typeId == TypeIds_BoolArray)
```

which means that if it was a `CharArray` or `BoolArray` (and not a native array) the condition would succeed, while now it would fail.

Reply:
That is true; my bad. Will revert this. Thanks for catching this.